docs(atomic-angular): add README for atomic angular (#1722)
coveo/ui-kit: packages/atomic-angular/README.md

# Atomic Angular

An Angular component library for building modern UIs interfacing with the Coveo platform. Atomic Angular is a wrapper around the core [Atomic](https://docs.coveo.com/en/atomic/latest/) web component library.

The integration of Angular-based projects with Web based components can be tricky. This project is meant to address this issue, making it possible to use Atomic web-components in a manner that feels more natural to developers familiar with Angular.

## Installation

`npm i @coveo/atomic-angular`

## Usage

> Since Atomic React is built on top of the core [Atomic](https://docs.coveo.com/en/atomic/latest/) web-components library, the vast majority of concepts that apply to core Atomic will apply to Atomic Angular.

Author: You found my copy paste ;)

  ngAfterViewInit(): void {
    this.atomicTextValueAttribute = this.val;
  }

  private get atomicTextValueAttribute() {
    if (!this.atomicText) {
      return '';
    }
    return this.atomicText['el'].getAttribute('value') as string;
  }

  private set atomicTextValueAttribute(v: string) {
    if (!this.atomicText) {
      return;
    }
    this.z.runOutsideAngular(() => {
      this.atomicText!['el'].setAttribute('value', v);
    });
  }
}
```

```html
<!-- field-label.component.html -->
<atomic-text #atomictext></atomic-text>
```

```html
<!-- Example usage in app.component.html -->

<app-field-label label="My label"> </app-field-label>
```

In the above example, we can see that we need to annotate `get label()` and `set label()` with the `@Input()` angular decorator.

When the label is set in the app component (`app.component.html`), we arrange for it to be passed down and propagated to `<atomic-text>` using `atomicText.el.setAttribute`. This ensure that that web component will properly receive the property without any modification of the property by the Angular rendering engine.

The reference to `atomicText` is then obtained with the `ViewChild('atomictext')` decorator.

> `atomictext` is the reference set in the HTML template (`<atomic-text #atomictext></atomic-text>`).

Reviewer: ```suggestion
`atomictext` is the reference set in the `<atomic-text #atomictext></atomic-text>` HTML template.
```

> Since that reference will only exists on `ngAfterViewInit`, we need to code defensively against undefined reference.

Reviewer: ```suggestion
Since that reference will only exist on `ngAfterViewInit`, we code defensively against undefined references.
```

## Installation

`npm i @coveo/atomic-angular`

## Usage

Since Atomic Angular is built on top of the core [Atomic](https://docs.coveo.com/en/atomic/latest/) web-components library, the vast majority of concepts that apply to core Atomic will apply to Atomic Angular.

However, there are still some special considerations.

## Static Assets - Languages and SVGs

For performance reasons, the generated JavaScript bundle does not automatically include static assets that are loaded on demand. This impacts language support, as well as the use of included SVG icons.

It is mandatory that you make available external assets distributed with Atomic Angular by including them in the public directory of your app. Without this, for example, labels in the app will appear as temporary placeholders.

The location of the public directory depends on how you build, configure and distribute your app.

For example, for any project created with [Angular CLI](https://angular.io/cli), this would mean copying language and icon assets to the root of the source directory

```
cp -r node_modules/@coveo/atomic-angular/assets src/assets src/assets
cp -r node_modules/@coveo/atomic-angular/lang src/lang
```

> It is important to respect the folder hierarchy, with SVG icons under the `assets` subdirectory, and labels and languages under the `lang` subdirectory.

Author: I've added this section and the `Including default Coveo Theme` section since I had forgotten to talk about those concepts @jpmarceau

> Once this is done, these folders must be configured as assets folder in the application.

Reviewer: ```suggestion
Once this is done, these folders must be configured as asset folders in the application.
```

> We can do so using the `angular.json` configuration file.

Reviewer: ```suggestion
You can do so using the `angular.json` configuration file.
```

Reviewer: question: is using `angular.json` the only way to do this? If so, then I'd just write:
```suggestion
Do so using the `angular.json` configuration file.
```

```json
"build": {
  // [...] omitted for brevity
  "options": {
    "assets": ["src/favicon.ico", "src/assets", "src/lang"],
  },
},
```

> ## Including default Coveo Theme

Reviewer: ```suggestion
## Including the Default Coveo Theme
```

> To include the default Coveo theme, you can use the `angular.json` configuration file.

Reviewer: Similarly to above, if using the `angular.json` file is the only way to do this, then:
```suggestion
To include the default Coveo theme, use the `angular.json` configuration file.
```
